### ISSUE TITLE
Incorrect compiler error: class 'Bar' has no member 'values'

### ISSUE THREAD

**Issue by heuermh** [ccrma/chuck#347]
The following used to work previous to ChucK version 1.5.1.0

```bash
$ cat Foo.ck
class Foo
{
  Bar @ bar;

  fun void run()
  {
    bar.values[0] @=> Object rv;
  }
}
class Bar
{
  Object values[10];
}

$ chuck Foo.ck
Foo.ck:7:9: error: class 'Bar' has no member 'values'
[7]     bar.values[0] @=> Object rv;
            ^
Foo.ck: ...in function 'void Foo.run()'
```

**Comment by gewang** [comment]
thank you for reporting; these are big issues and we will prioritize boost fixing them; looking into this and #348 today

**Comment by gewang** [comment]
This revealed an issue going back to 1.5.0.8, specifically when we added `auto`, which deferred certain variable declarations, including those of class variables; this has been addressed to reinstate support for "declare-anywhere" class defs and allow `auto` type, with the caveat (for now) that `auto` cannot be used to declare top-level class variables (ones that can be accessed from outside). Really glad you caught this. Thank you.

The code change was committed in 0ece499, with new unit tests in 277216f

Hopefully this fixes this issue; please let us know if you run into issues with this or anything else! Once we can verify this and un-break LiCK build (sorry about that!), we will put out a 1.5.1.1 patch release. We have also started working on an extended test set (real-world codebases, beyond our unit tests) for verifying each release

**Comment by heuermh** [comment]
Thank you, @gewang!  Closing, fixed as of 1.5.1.1 release
